src/Hosting.Services/HostBuilderExtensions.cs

```diff
@@ -60,7 +61,16 @@ public static IServiceCollection AddOmexServiceFabricDependencies<TContext>(this
 
 			collection.TryAddSingleton<IServiceContext, OmexServiceFabricContext>();
 			collection.TryAddSingleton<IExecutionContext, ServiceFabricExecutionContext>();
+
+			collection.AddLogging();
+
+			collection.TryAddTransient<IServiceContext, EmptyServiceContext>();
```

Comment:
Why register these types as transient after registering the interfaces with different types (and as Singleton) above?
Suggest adding a comment here.

Reply:
this code is copied from `Logging/ServiceCollectionExtensions.cs`. Do you suggest changing to TryAddSingleton instead (which makes sense) ?

Reply:
Repeated calls to TryAddXXX will not override the first registration.

Reply:
see my explanation here: https://github.com/microsoft/Omex/pull/673/files#r1598767613 in this file src/Logging/ServiceCollectionExtensions.cs. 
If consumers do not call `AddOmexLogging()`, they will lose EmptyServiceContext, BaseExecutionContext, LoggerExternalScopeProvider. 
So this is the place to keep them so as not to break other code.

Reply:
This code in its current state doesn't make sense. You have to figure out the scenarios where EmptyServiceContext and BaseExecutionContext get used instead of OmexServiceFabricContext and ServiceFabricExecutionContext, and make sure those scenarios continue working.

OR

Be less aggressive with obsoletion, mark just those classes and methods which you 100% know will get deleted, and leave the rest untouched.